src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -81,37 +81,447 @@ constexpr const ad_utility::Last<Current, Others...>* unwrapVariant(
+    for (auto& tuple : tuples) {
+      triples.push_back(
+          TripleWithPropertyPath{subject, tuple.first, tuple.second});
+    }
+  } else if (ctx->triplesNodePath()) {
+    throwCollectionsAndBlankNodePathsNotSupported(ctx->triplesNodePath());
+  }
+  return triples;
+}
+
+// ___________________________________________________________________________
+std::optional<PathTuples> SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::PropertyListPathContext* ctx) {
+  if (ctx->propertyListPathNotEmpty()) {
+    return visitTypesafe(ctx->propertyListPathNotEmpty());
+  } else {
+    return std::nullopt;
+  }
+}
+
+// ___________________________________________________________________________
+PropertyPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::VerbPathContext* ctx) {
+  PropertyPath p = visitTypesafe(ctx->path());
+  p.computeCanBeNull();
+  return p;
+}
+
+// ___________________________________________________________________________
+Variable SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::VerbSimpleContext* ctx) {
+  return visitTypesafe(ctx->var());
+}
+
+// ___________________________________________________________________________
+ObjectList SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::ObjectListPathContext* ctx) {
+  // The second parameter is empty because collections and blank not paths,
+  // which might add additional triples, are currently not supported.
+  return ObjectList{visitVector<VarOrTerm>(ctx->objectPath()), {}};
+}
+
+// ___________________________________________________________________________
+PropertyPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::PathContext* ctx) {
+  return visitTypesafe(ctx->pathAlternative());
+}
+
+// ___________________________________________________________________________
+PropertyPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::PathAlternativeContext* ctx) {
+  auto paths = visitVector<PropertyPath>(ctx->pathSequence());
+
+  if (paths.size() == 1) {
+    return paths[0];
+  } else {
+    return PropertyPath::makeAlternative(std::move(paths));
+  }
+}
+
+// ___________________________________________________________________________
+PropertyPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::PathSequenceContext* ctx) {
+  auto paths = visitVector<PropertyPath>(ctx->pathEltOrInverse());
+
+  if (paths.size() == 1) {
+    return paths[0];
+  } else {
+    return PropertyPath::makeSequence(std::move(paths));
+  }
+}
+
+// ___________________________________________________________________________
+ad_utility::sparql_types::VarOrPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::VerbPathOrSimpleContext* ctx) {
+  if (ctx->verbPath()) {
+    return visitTypesafe(ctx->verbPath());
+  } else if (ctx->verbSimple()) {
+    return visitTypesafe(ctx->verbSimple());
+  }
+  AD_FAIL()  // Should be unreachable.
```

Comment:
I am considering a `visitAlternative`-abstraction like,
```
return `visitAlternative<VarOrPath>(ctx->verbPath(), ctx->verbSimple());
```
Which expands to this code pattern which we have several times (but I am not sure how often:))